tests/test_location_pi.py

```diff
@@ -0,0 +1,508 @@
+# -*- coding: utf-8 -*-
+
+from datetime import datetime, timedelta
+import time
+
+from . import common
+
+from odoo.exceptions import ValidationError
+
+
+ODOO_DATETIME_FORMAT = '%Y-%m-%d %H:%M:%S'
+
+
+class MockPIOutcome:
+    def __init__(self, got_changes):
+        self._got_changes = got_changes
+
+    def got_inventory_changes(self):
+        return self._got_changes
+
+
+class TestLocationPI(common.BaseUDES):
+
+    @classmethod
+    def setUpClass(cls):
+        super(TestLocationPI, cls).setUpClass()
+        Location = cls.env['stock.location']
+        User = cls.env['res.users']
+
+        user_warehouse = User.get_user_warehouse()
+        cls.picking_type_pick = user_warehouse.pick_type_id
+        cls.datetime_tolerance = timedelta(minutes=1)
+
+        loc_id = 9834247
+
+        while Location.browse(loc_id).exists():
+            if loc_id > 9835147:
+                # done 100 attempts, that's enough
+                cls.fail("Failed to determine an unknown location; "
+                         "please use an empty database for testing")
+            else:
+                loc_id += 9
+
+        cls.unknown_location_id = loc_id
+
+    def setUp(self):
+        super(TestLocationPI, self).setUp()
+        Package = self.env['stock.quant.package']
+
+        self.package_one = Package.get_package("test_pi_package_one",
+                                               create=True)
+        self.package_two = Package.get_package("test_pi_package_two",
+                                               create=True)
+
+    def test01_validate_pi_request_success(self):
+        """ No error is raised for a good PI request """
+        req = {
+            'location_id': self.test_location_01.id,
+            'pi_count_moves': [
+                {
+                    'quants_ids': [1, 2, 3],
```

Comment:
How `quants_ids` ended up? quant_ids or quants_ids?

Reply:
it's a typo, it should be `quant_ids`, but it's not used by the method under test

Reply:
Ok, but can we fix it? This might confuse the reader.